armory/scenarios/carla_object_detection.py

```diff
@@ -29,46 +30,46 @@ def load_dataset(self):
             raise ValueError("batch_size must be 1 for evaluation.")
         super().load_dataset(eval_split_default="dev")
 
-    def run_benign(self):
-        x, y = self.x, self.y
-        y_object, y_patch_metadata = y
+    def next(self):
+        x, y = next(self.test_dataset)
+        i = self.i + 1
+        self.i, self.x, self.y = i, x, y
+        self.y_object = [y[0]]
+        self.y_patch_metadata = [y[1]]
+        self.y_pred, self.y_target, self.x_adv, self.y_pred_adv = None, None, None, None
```

Comment:
This could be simplified to:
```
    def next(self):
        super().next()
        self.y_object = [self.y[0]]
        self.y_patch_metadata = [self.y[1]]
```